### PR TITLE
changed video source for better resolution

### LINKED ISSUE #1489: Insufficient resolution of some videos within a ZIM
In this ZIM http://iiab.me/downloads/wp_en_mdwiki_2021-06.zim

For the article on gout the first video has a great resolution while the second has a poor resolution

This is the first video https://commons.wikimedia.org/wiki/File:En.Wikipedia-VideoWiki-Gout.webm

This is the second https://commons.wikimedia.org/wiki/File:Gout.webm

And this is the page it is built from https://mdwiki.org/wiki/Gout

We are needing a method to set the minimum resolution of a video instead of it just choosing the lowest possible resolution per https://github.com/openzim/mwoffliner/issues/1442.

The poor resolution issue for Osmosis videos persists even when these videos are in the same spot as the good quality non Osmosis videos. See amnesia as an example.

### PR DESCRIPTION
**Pull request by MananJethwani**
this fixes #1489

### CODE REVIEW

File: src/util/saveArticles.ts [hunk 1]
```diff
@@ -412,13 +412,26 @@ export async function treatVideo(mw: MediaWiki, dump: Dump, srcCache: KVS<boolea
     }
 
     videoSources = videoSources.sort(widthXHeightSorter);
+    const videoElArea = (videoEl.getAttribute('height') || 0) * (videoEl.getAttribute('width') || 0);
```
Comment (kelson42) [inline src/util/saveArticles.ts:415]: Here we need a comment explaining a bit the problem and the solution

Comment (kelson42) [inline src/util/saveArticles.ts:415]: I don't understand really the computation with width*heigth. Considering that all the videos have the same ratio (width/height), I think just comparing the widths should be enough.

File: src/util/saveArticles.ts [hunk 2]
```diff
@@ -412,13 +412,26 @@ export async function treatVideo(mw: MediaWiki, dump: Dump, srcCache: KVS<boolea
...
+        return true;
+    });
 
-    const sourcesToRemove = videoSources.slice(1); // All but first
+    if(!sourceEl) {
+        sourceEl = sourcesToRemove[0];
+        sourcesToRemove = sourcesToRemove.slice(1);
+    }
 
     sourcesToRemove.forEach(DU.deleteNode);
```
Comment (kelson42) [inline src/util/saveArticles.ts:433]: This is simply wrong, `sourcesToRemove` is not always defined.

**Review by kelson42 — CHANGES_REQUESTED** [review]
What is the difference beetween `data-width` and `data-file-width`, it does not looks good to take both of them?

This should be tested with an automated test.

**Review by kelson42 — CHANGES_REQUESTED** [review]
I think the whole approach can be simplified a lot:
* Remove `widthXHeightSorter` because we don't need that anymore as we don't take the small resolution anymore
* Just go through all the `source` node element and either keep the node (because this is a better choice than the previous) or remove it. At the end it should be only one left.
* I'm not even sure you need a dedicated function, so probably `videoSourceElement` could be deleted as well. If you want to isolate things in a dedicated function just create a function `videoSourceUrlPicker()` (current name `videoSourceElement` is pretty cryptic as well).

You should be able to do all of that in twice less lines than what we have today.... at least.

**Comment by kelson42** [comment]
@tim-moody Does it do the job properly now?

**Comment by tim-moody** [comment]
@WikiDocJames and I will try to test.

**Review by kelson42 — APPROVED** [review]
LGTM now

**Comment by kelson42** [comment]
Will merge this PR, if a problem is still occuring, please open a new ticket.

**Comment by WikiDocJames** [comment]
Okay we tested it 16 hours ago. But looks like you have just pushed the fix 3 hour ago. So will test again this am...

**Comment by tim-moody** [comment]
@kelson42 I didn't notice that the changes weren't merged. Am I now to test with the latest docker image or a nightly or by building a branch?

**Comment by tim-moody** [comment]
docker pull openzim/mwoffliner:latest says Image is up to date for openzim/mwoffliner:latest

**Comment by tim-moody** [comment]
@kelson42 when do we expect this fix to be in the docker image?

**Pull request merged**